src/types.cpp

```diff
@@ -295,7 +319,7 @@ SizedType CreatePointer(const SizedType &pointee_type)
 
 SizedType CreateRecord(size_t size, const std::string &name)
 {
-  auto ty = SizedType(Type::record, size);
+  auto ty = SizedType(Type::record, size / 8);
```

Comment:
I'll leave the previous comments up for my own benefit later, but why go through all the trouble of passing in bits if it gets stored as bytes later? If you really want these semantics, maybe name the constructor parameter `size_bits` to be extra clear.

Reply:
Yes that can  do with a  bit of a cleanup, I'll give it a go

I think that having  sizes in bits eventually makes more sense. As booleans will be impossible otherwise